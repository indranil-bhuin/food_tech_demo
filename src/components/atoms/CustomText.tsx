import React from 'react';
import { StyleSheet, Text } from 'react-native';

const CustomText = ({ children, style, color}) => {
  return <Text style={[style, {...styles.customText, color}]}>{children}</Text>;
};

const styles = StyleSheet.create({
  customText: {
    color: 'white',
    fontSize: 24,
    fontFamily: 'Nunito-Black'
  }
})
export default CustomText;
