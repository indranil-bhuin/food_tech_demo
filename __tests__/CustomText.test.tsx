import React from 'react';
import renderer from 'react-test-renderer';
import CustomText from '../src/components/atoms/CustomText';

describe('CustomText', () => {
  it('renders text with proper styling', () => {
    const tree = renderer
      .create(<CustomText color="#000000">Hello, World!</CustomText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
