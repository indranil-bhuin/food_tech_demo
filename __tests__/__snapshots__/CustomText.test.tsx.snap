// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText renders text with proper styling 1`] = `
<Text
  style={
    [
      undefined,
      {
        "color": "#000000",
        "fontFamily": "Nunito-Black",
        "fontSize": 24,
      },
    ]
  }
>
  Hello, World!
</Text>
`;
